go/tao/auth/lexer.go

```diff
@@ -81,6 +79,16 @@ var (
 	tokenEOF       = token{itemEOF, nil}
 )
 
+var prinTokens = map[token]bool{
```

Comment:
We would need to add a context to the parser. I'm thinking a struct called Lexicon with methods IsPrinToken(i token), AddPrinTokens(keywords ...string), and a constructor NewLexiconFromPrinTokens(keywords ...string). We would need to instantiate a Lexicon in the policy datalog policy guard, as well as anywhere where datalog parsing is done. Then I think we would need to add the set of valid prinTokens to the domain template.